WebDevStudios/wd_s
# Critical CSS

### DESCRIPTION

This separates some critical CSS partials and compiles them into their own file, which is then added inline in the `<head>`.

What I did:
- Moved `menus.scss`, `mobile-menus.scss`, and `header.scss` from their original locations to a `src/scss/critical/` directory.
- Added multiple entry points to our `webpack.config.js` file to produce multiple build files for `index.css` and `critical.css`
- Inlined the styles in the `<head>`

What I'd like to get some :eyes: on:
- Make sure this is the cleanest and safest way to include this partial
- Add to/adjust what's being compiled into `critical.css` as needed

### SCREENSHOTS
<img width="1151" alt="image" src="https://user-images.githubusercontent.com/954724/133835994-162f4c50-3c79-40de-ba6e-ec60e285d76e.png">
<img width="1635" alt="image" src="https://user-images.githubusercontent.com/954724/133836267-1d552f6e-d216-4b20-838e-be8cb7c8d506.png">

### OTHER


### STEPS TO VERIFY
1. Checkout
2. Run `npm run build`
3. Verify the inline styles are in the `<head>` and the header & menus are looking good
4. Verify the method of including these styles is safe & secure

### DOCUMENTATION

Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

I can see us wanting to update around Critical CSS paths and what should be included there, so I'll start looking at the existing docs to see where this could best fit.
